crates/core/tedge_agent/src/main.rs

```diff
@@ -1,7 +1,25 @@
+use cap::Cap;
 use clap::Parser;
+use std::alloc;
+use std::time::Duration;
+
+#[global_allocator]
+static ALLOCATOR: Cap<alloc::System> = Cap::new(alloc::System, usize::MAX);
 
 #[tokio::main]
-async fn main() -> Result<(), anyhow::Error> {
+async fn main() -> anyhow::Result<()> {
```

Comment:
We have to consider deprecating this `main` and enforcing the use of a `tedge -agent` as a symlink to `tedge`.

Reply:
Here is a ticket for that: https://github.com/thin-edge/thin-edge.io/issues/2696